script/grant_revoke_gem_authority.rb

```diff
@@ -1,166 +1,142 @@
 ]
-# Email addresses that are known not to be registered at rubygems.org
-SKIP_EMAILS = [
-  'geisler@stanford.edu',
-  'dlrueda@stanford.edu',
-  'jgreben@stanford.edu',
-  'lmcglohon@gmail.com',
-  'ssklar@stanford.edu',
-  'tony.zanella@gmail.com',
-  'blalbritton@gmail.com',
-  'dgcliff@northeastern.edu',
-  'jessica@ucsd.edu'
-]
+
 VERBOSE = ENV.fetch('VERBOSE', false)
 
-puts "(Hang in there! This script takes a couple minutes to run.)"
+# See https://guides.rubygems.org/rubygems-org-api/#rate-limits I'm
+# opting for the slower 0.2 so as to not come close to the rate limit.
+GEM_SLEEP_INTERVAL = 0.2
 
-github = Github.new(oauth_token: AUTHORIZATION_TOKEN, auto_pagination: true)
+$stdout.puts "Hang in there! This script takes a couple minutes to run."
 
-# Get the ID of the GitHub-provided "Owners" team from the samvera org
-# We don't currently grant gem ownership to the folks in the other two orgs
-# (This just preserves the prior behavior.)
-owner_team_id = github.orgs.teams.list(org: 'samvera').select { |team| team.name == 'Admins' }.first.id
-owners = github.orgs.teams.list_members(owner_team_id)
-# Start with the prior (known to work) list of email addresses
-committer_map = KNOWN_COMMITTER_EMAIL_ADDRESSES.dup
-owners.each do |owner|
-  user = github.users.get(user: owner.login)
-  # Move along if the user doesn't have an email addy or if there's already an entry in the map
-  next if !user.respond_to?(:email) || user.email.nil? || user.email.empty? || !committer_map[user.login].nil?
-  committer_map[user.login] = user.email
-end
-committer_emails = committer_map.values.sort.uniq
+github = Github.new(oauth_token: AUTHORIZATION_TOKEN, auto_pagination: true)
 
 # Keep track of things
 @errors = []
 @bogus_gem_names = []
 @gem_names = HANGERS_ON
 
-def exists?(name)
-  system("gem owner #{name} > /dev/null 2>&1")
-end
-
-def replace_known_mismatch(name)
-  KNOWN_MISMATCHED_GEM_NAMES.fetch(name, name)
-end
+@definitive_committers = File.read(
+  File.expand_path("../../data/definitive-rubygems-authors-list.txt", __FILE__)
+).split("\n").map { |line| line.sub(/ *#.*$/, "") }.select { |line| !line.empty? }
 
 ORGANIZATION_NAMES.each do |org_name|
   github.repos.list(org: org_name).each do |repo|
-    name = replace_known_mismatch(repo.name)
-    if exists?(name)
+    # Replace known mismatched names
+    name = KNOWN_MISMATCHED_GEM_NAMES.fetch(repo.name, repo.name)
+
+    # Let's move on to any gem that we know will be a false positive.
+    next if FALSE_POSITIVES.include?(name)
+
+    # TODO: We are twice calling `gem owner` for information. We could
+    # rework the logic to only call it once.
+    sleep(GEM_SLEEP_INTERVAL)
+    if system("gem owner #{name} > /dev/null 2>&1")
       @gem_names << name
     else
       @bogus_gem_names << repo.full_name
     end
   end
 end
 
-def gem_owner_with_error_check(gemname, params)
-  command = "gem owner #{gemname} #{params}"
-  puts "running: #{command}" if VERBOSE
+OWNERSHIP_ACTIONS = {
```

Comment:
I have a slight preference for declaring constants together near the top of a file but I'm not sure this file already respects this preference, so :man_shrugging: